crates/wasmtime/src/compile/code_builder.rs

```diff
@@ -177,10 +206,367 @@ impl<'a> CodeBuilder<'a> {
         }
     }
 
-    pub(super) fn get_wasm(&self) -> Result<&[u8]> {
-        self.wasm
-            .as_deref()
-            .ok_or_else(|| anyhow!("no wasm bytes have been configured"))
+    /// Get the Wasm to be compiled.
+    ///
+    /// When using compile-time builtins, compose the builtins and the guest
+    /// Wasm together, before returning the Wasm.
+    pub(super) fn get_wasm(&self) -> Result<Cow<'_, [u8]>> {
+        let wasm = self
+            .wasm
+            .clone()
```

Comment:
This `.clone()` I think may not be desired since that clones the owned `Cow` if it's owned. I think you still want the `as_deref` from before?